go-controller/pkg/ovn/pods.go

```diff
@@ -326,8 +335,12 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	}
 
 	if annotation == nil {
-		gwIfAddr := util.GetNodeGatewayIfAddr(nodeSubnet)
-		routes, gwIP, err := getRoutesGatewayIP(pod, gwIfAddr)
+		nsInfo, err := oc.waitForNamespaceLocked(pod.Namespace)
+		if err != nil {
+			return err
+		}
+		defer nsInfo.Unlock()
```

Comment:
@rcarrillocruz actually, I don't think we want defer here; that'll lock the namespace object for much longer than we need including while writing annotations to the kube apiserver which is expensive.  Instead, I would make a separate function:

```
func (oc *Controller) waitForNamespaceAnnotations(ns string) map[string]string {
	nsInfo, err := oc.waitForNamespaceLocked(pod.Namespace)
	if err != nil {
		return err
	}
	defer nsInfo.Unlock()
	return nsInfo.annotations
}
```

And just call that from addLogicalPort().

Reply:
Per https://github.com/ovn-org/ovn-kubernetes/pull/1209#discussion_r414839721https://github.com/ovn-org/ovn-kubernetes/pull/1209#discussion_r414839721 still have the separate function, but just make it return nsInfo.externalGatewayIP or whatever name you call it.